apps/api/src/goods-receipts/goods-receipts.controller.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { GoodsReceiptsController } from './goods-receipts.controller';
+import { GoodsReceiptsService } from './goods-receipts.service';
+
+describe('GoodsReceiptsController', () => {
+  let controller: GoodsReceiptsController;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [GoodsReceiptsController],
+      providers: [GoodsReceiptsService],
+    }).compile();
+
+    controller = module.get<GoodsReceiptsController>(GoodsReceiptsController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+});
```

Comment:
What is your stance on testing, automated or otherwise? In either case, how much do you test and what types of test do you reccommend?

Reply:
I like a healthy mix of unit and integration tests. Unit tests are ideal for validating individual functions and business logic, such as the 3-way match code I wrote. Integration tests are great for things like ensuring API endpoints return expected results and the service code modifies the database correctly. If you have comprehensive integration tests, you shouldn't need unit tests for every function. Code coverage should be measured and all automated tests should be part of the CI/CD pipeline. 

For manual testing, I think using exploratory testing and UAT makes sense. I have found that exploratory testing works great when done by a UX designer. UAT should be done to gather feedback from real users before production releases. 

Additionally, performance and load testing should be performed as your application scales.

Reply:
What is the stance on testing at GoodDay?